### PR TITLE
fixes solid

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
## Changes

- closes https://github.com/withastro/adapters/issues/209

## Testing

- manual tested + existing solid test

## Docs

- bug fix, no docs needed

### CODE REVIEW

**Comment by alexanderniebuhr** [comment]
!preview cf-10-solid

File: packages/cloudflare/src/index.ts [hunk 1]
```diff
@@ -191,6 +186,13 @@ export default function createIntegration(args?: Options): AstroIntegration {
 						...vite.define,
 					};
 				}
+				if (target === 'client') {
+					vite.resolve ||= {};
+					vite.resolve.conditions ||= [];
+					vite.resolve.conditions = vite.resolve.conditions.filter(
+						(c) => c !== 'workerd' && c !== 'worker'
+					);
+				}
```
Comment (ematipico) [inline packages/cloudflare/src/index.ts:195]: It's always good to leave comments that explain these... workaround?

**Review by ematipico — APPROVED** [review]

File: packages/cloudflare/src/index.ts [hunk 2]
```diff
@@ -170,6 +162,9 @@ export default function createIntegration(args?: Options): AstroIntegration {
 						}
 					}
 
+					vite.resolve.conditions ||= [];
+					vite.resolve.conditions.push('workerd', 'worker');
```
Comment (alexanderniebuhr) [inline packages/cloudflare/src/index.ts:166]:
```suggestion
					// We need those conditions, previous these conditions where applied at the esbuild step which we removed
					// https://github.com/withastro/astro/pull/7092
					vite.resolve.conditions.push('workerd', 'worker');
```

File: packages/cloudflare/src/index.ts [hunk 3]
```diff
@@ -191,6 +188,13 @@ export default function createIntegration(args?: Options): AstroIntegration {
 						...vite.define,
 					};
 				}
+				if (target === 'client') {
```
Comment (alexanderniebuhr) [inline packages/cloudflare/src/index.ts:191]:
```suggestion
				// we thought that vite config inside `if (target === 'server')` would not apply for client
				// but it seems like the same `vite` reference is used for both
				// so we need to reset the previous conflicting setting
				// in the future we should look into a more robust solution
				if (target === 'client') {
```

**Pull request merged**